Remove PHP Dependency

AnnaDodson opened the pull request:
- All page fragments are now html
- Site is built through Grunt which concats the page fragments in order
- Add task to serve the built site on localhost through running `grunt serve`
- Method to generate random quote in footer now JavaScript method in
  `app.js`
- Update README to new build details
- `index.html` is now a generated file and added to the gitignore

Closes #29

Hbrinj commented:
Hey, from the brief look this looks promising. Just a little thing can you split the commits up logically please, this will be useful if something isn't working quite right we can bisect to the exact commit etc.

Commit:
Replace PHP Dependency with Grunt

- Split `index.html` into `head.html` and `end.html` and moved into `templates` directory
- All page fragments changed to html from php
- Grunt concats and minifies all html page fragments and outputs to `index.html`
- Changed quote function from php to javascript, moved into `js/app.js`
- `Index.html` and `development.html` are now generated files, added to gitignore
- Update `README.md`

Commit:
Enable Grunt Serve for localhost

- Add `grunt serve` to serve the site on localhost and watch changes
  easily
- Included the plugin to `package.json`
- Update `README.md` instructions to new build details

AnnaDodson commented:
Thanks @Hbrinj! I've split into three commits, let me know for anything else :+1:

Hbrinj commented on templates/footer.html:
```diff
@@ -0,0 +1,21 @@
+
+<script src="main.min.js"></script>
+
+</body>
```
Can this and the end HTML tag be moved out to the index file? what if i exclude the footer file.

AnnaDodson replied:
I see your point. The index file is now auto generated from Grunt but I could move it into a new `end.html` file so it would just sit at the end of the concatenated file? The footer file could then easily be removed or changed.

Hbrinj commented:
Hi Anna,
maybe merge build side and removal of php dependency. Thanks
